CHANGES.md

```diff
@@ -15,17 +15,125 @@ When the path to Java is known, SAW can automatically add system-related
+
+* The verification code initializes all JVM classes at start so that
+  initializers don't run at arbitray intermediate points and and clobber
+  static field values specified in preconditions. This means, however,
+  that any proofs about Java code are under the assumption that all
+  class initializers have run before the method under analysis executes.
+
+Now that the Crucible-based verification infrastructure is sufficiently
+expressive and performant, we have removed all dependencies on the old
+`jvm-verifier` library.
+
+On the LLVM side, SAWScript includes a variety of new functions for
+writing specification blocks:
+
+* The `llvm_struct_type` and `llvm_packed_struct_type` functions each
+  construct an LLVM struct type from a list of other LLVM types. This is
+  not to be confused with the existing `llvm_struct` function, which
+  takes a string as an argument and returns the corresponding alias type
+  (which is often, but not necessarily, defined as a struct type).
+
+* To avoid confusion, a new `llvm_alias` function now exists, and
+  `llvm_struct` is now a synonym for `llvm_alias`. The `llvm_struct`
+  function continues to be available for now.
+
+* The `llvm_pointer : LLVMType -> LLVMType` function allows construction
+  of arbitrary LLVM pointer types.
+
+* Two new functions, `llvm_points_to_at_type` and
+  `llvm_conditional_points_to_at_type`, mirror `llvm_points_to` and
+  `llvm_conditional_points_to`, but cast the pointer to a different
+  type. This may be useful when reading or writing a prefix of larger
+  array, for example.
+
+* Support for using ABC as an external process is more complete:
+
+    * SAW can now generate Verilog with multiple outputs (from `Term`
+      values that have tuple or vector result types, for example).
+
+    * The new commands `write_aig_external` and `write_cnf_external`
+      generate AIG and CNF files, respectively, by first writing Verilog
+      and then using the available `abc` executable to bit-blast to the
+      lower-level representation. Corresponding proof tactics,
+      `offline_aig_external` and `offline_cnf_external` also exist.
+
+These changes are in preparation for removing the linked-in copy of ABC
+in a future release.
+
+The `saw-remote-api` RPC server and associated Python client now have
+more complete support for LLVM verification, including:
+
+* More complete points-to declarations, matching what is currently
+  available in SAWScript.
+
+* Support for more provers, including the full range of SBV-based and
+  What4-based provers available in SAWScript.
+
+* Support for ghost variables.
+
+* Support for assuming LLVM contracts directly (rather than the previous
+  behavior which would temporarily assume that failed verifications
+  succeeded to determine whether higher-level verifications would still
+  succeed).
+
+* Support for global variables and initializers.
+
+* Support for null pointers.
+
+* Support for array value literals.
```

Comment:
Some other PRs from the now-defunct `galois-py-toolkit` repo that might be worth mentioning:

* Support for specifying alignment in `alloc()`: https://github.com/GaloisInc/galois-py-toolkit/pull/17
* Support for accessing pointers by element index or field name: https://github.com/GaloisInc/galois-py-toolkit/pull/9

Reply:
Ah, thanks! I'll include those.